src/search/command_line.cc

```diff
@@ -129,11 +188,11 @@ static shared_ptr<SearchEngine> parse_cmd_line_aux(
 
 
 shared_ptr<SearchEngine> parse_cmd_line(
-    int argc, const char **argv, options::Registry &registry, bool dry_run, bool is_unit_cost) {
+    int argc, const char **argv, bool is_unit_cost) {
     vector<string> args;
     bool active = true;
     for (int i = 1; i < argc; ++i) {
-        string arg = sanitize_arg_string(argv[i]);
+        string arg = argv[i];
 
         if (arg == "--if-unit-cost") {
```

Comment:
Our new `let` syntax lose functionality regarding `--if-unit-cost`. Previously you could predefine an evaluator/landmark factory based on whether the given task has unit costs or not and then use one of the predefined in the search, e.g.
```
--if-unit-cost --evaluator h=ff() --if-not-unit-cost --evaluator h=lmcut() --always --search "astar(h)"
```
Now it would look like this (of course imagine a longer configuration than astar(h)):
```
--if-unit-cost --search "let(h, ff(), astar(h))" --if-non-unit-costs --search "let(h, ff(), astar(h))"
```

It would be strange to mark the old syntax as deprecated when the new one has less power. Using the old syntax with our automatic translation allows to combine the best of both worlds, e.g.
```
--heuristic h=ff(let(..)...) --search astar(h)
->
--search let(h, ff(let(..)...), astar(h)
```
the automatic translation does not care about the actual value of the variable defined.

As stated above would be strange to remove something in th efuture with allows more than what we want to switch to.
Option A: So be it
Option B: Introduce --predefine instead of --evaluator/heuristic/landmark which we do not mark as deprecated and explicitly tell the user this is transformed into let syntax
Option C: Introduce an `if` in our syntax